octodns/provider/selectel.py

```diff
@@ -0,0 +1,278 @@
+        if resp.status_code == 401:
+            raise SelectelAuthenticationRequired(resp.text)
+        elif resp.status_code == 404:
+            return {}
+        resp.raise_for_status()
+        if method == 'DELETE':
+            return {}
+        if resp.json():
+            return resp.json()
+
+        self.log.debug('_request: empty response')
+        return {}
+
+    def _apply(self, plan):
+        desired = plan.desired
+        changes = plan.changes
+        self.log.debug('_apply: zone=%s, len(changes)=%d', desired.name,
+                       len(changes))
+
+        zone_name = desired.name[:-1]
+
+        for change in changes:
+            class_name = change.__class__.__name__
+            getattr(self, '_apply_{}'.format(class_name).lower())(zone_name,
+                                                                  change)
+
+    def _apply_create(self, zone_name, change):
+        new = change.new
+        params_for = getattr(self, '_params_for_{}'.format(new._type))
+        for params in params_for(new):
+            self.create_record(zone_name, params)
+
+    def _apply_update(self, zone_name, change):
+        self._apply_delete(zone_name, change)
+        self._apply_create(zone_name, change)
+
+    def _apply_delete(self, zone_name, change):
+        existing = change.existing
+        self.delete_record(zone_name, existing._type, existing.name)
+
+    def _params_for_multiple(self, record):
+        for value in record.values:
+            yield {
+                'content': value,
+                'name': record.fqdn,
+                'ttl': max(self.MIN_TTL, record.ttl),
+                'type': record._type,
+            }
+
+    def _params_for_single(self, record):
+        yield {
+            'content': record.value,
+            'name': record.fqdn,
+            'ttl': max(self.MIN_TTL, record.ttl),
+            'type': record._type
+        }
+
+    def _params_for_MX(self, record):
+        for value in record.values:
+            yield {
+                'content': value.exchange,
+                'name': record.fqdn,
+                'ttl': max(self.MIN_TTL, record.ttl),
+                'type': record._type,
+                'priority': value.preference
+            }
+
+    def _params_for_SRV(self, record):
+        for value in record.values:
+            yield {
+                'name': record.fqdn,
+                'target': value.target,
+                'ttl': max(self.MIN_TTL, record.ttl),
+                'type': record._type,
+                'port': value.port,
+                'weight': value.weight,
+                'priority': value.priority
+            }
+
+    _params_for_A = _params_for_multiple
+    _params_for_AAAA = _params_for_multiple
+    _params_for_NS = _params_for_multiple
+    _params_for_TXT = _params_for_multiple
+
+    _params_for_CNAME = _params_for_single
+
+    def _data_for_A(self, _type, records):
+        return {
+            'ttl': records[0]['ttl'],
```

Comment:
Based on the MIN/MAX elsewhere I believe you'll run into the same stuff I hit with the Cloudflare provider when there's a TTL configured on a record below the min (or above the max in this case.)

There's an `_include_change` method that allows the provider to look things over at the correct point and ignore things that only mis-match b/c of an unsupported ttl. You can see how that works for Cloudflare [here](https://github.com/github/octodns/blob/c45b7a5fb0c4e75987a866667de242ad78858fe1/octodns/provider/cloudflare.py#L193-L200). You'll probably need something very similar for this provider.